scripts/globals/znm.lua

```diff
@@ -16,27 +17,98 @@ require("scripts/globals/msg")
 require("scripts/globals/npc_util")
 require("scripts/globals/pankration")
 require("scripts/globals/utils")
+require("scripts/globals/znm_data")
 -----------------------------------
 
 xi = xi or {}
 xi.znm = xi.znm or {}
 
+--------------------------------------------------
+-- Sanraku's Interest and Recommended Fauna
+-- Applies bonuses to soul plate zeni-value
+--------------------------------------------------
+-- Called during JstMidnight tick
+xi.znm.UpdateSanrakusMobs = function()
+    SetServerVariable('[ZNM][Sanraku]Interest', math.random(#xi.znm.SANRAKUS_INTEREST))
+    SetServerVariable('[ZNM][Sanraku]Fauna', math.random(#xi.znm.SANRAKUS_FAUNA))
+end
+
+-- Get Sanraku's "Subject of Interest"
+xi.znm.getSanrakusInterest = function()
+    local interest = GetServerVariable('[ZNM][Sanraku]Interest')
+
+    -- Initialize the server var if it hasn't been already
+    if interest == nil or interest == 0 then
+        interest = math.random(#xi.znm.SANRAKUS_INTEREST)
+        SetServerVariable('[ZNM][Sanraku]Interest', interest)
+    end
+    return interest
+end
+
+-- Get Sanraku's "Recommended Fauna"
+xi.znm.getSanrakusFauna = function()
+    local fauna = GetServerVariable('[ZNM][Sanraku]Fauna')
+
+    -- Initialize the server var if it hasn't been already
+    if fauna == nil or fauna == 0 then
+        fauna = math.random(#xi.znm.SANRAKUS_FAUNA)
+        SetServerVariable('[ZNM][Sanraku]Fauna', fauna)
+    end
+    return fauna
+end
+
+--- Does this mob fall under Sanraku's current "Subject of Interest"?
+-- Some families had multiple famIDs (see Sea Monks), and some superfamIDs were too general (see elementals)
+-- In order to only need to use/store a single ID per mob, a bool was added to check what type of ID is stored
```

Comment:
Zeni calculation is a two step process.  Basic parameters must be captured at the time the picture is taken (hpp, distance, etc).  Bonus parameters (matching fauna or interest) can only be determined when the plate is traded.

The boolean for matching fauna or interest was introduced due to the wings code base having limitations on pulling mob information.  We did not have a passive way to generate or look up mob information, and considering instance mob generation required the instance the mob was spawned in - active generation was ruled out.   (read: I wasnt going to implement a full dynamic mob system or mob query system to get a value off a zeni plate)

Additionally, there are space limitations for m_extra, so getting down to just a few bytes of data was ideal.

In LSB, I was considering that we may be able to resolve this and return to storing mobID, which would simplify some use cases like the ASA mission which uses the soul trapper.  However, Dynamic Mobs (if I understand the impl correctly) would add complexity to that endeavor.  This could be solved by not allowing pictures of dynamic mobs but half the fun of soul trappers is seeing them go off in dynamis.